doc/developer-notes.md

```diff
@@ -2,41 +2,47 @@ Developer Notes
 ===============
 
 <!-- markdown-toc start -->
+<!-- markdownlint-disable no-emphasis-as-header -->
```

Comment:
I wouldn't be in favor of having to turn `markdownlint` warnings on and off throughout files. It's going to just look messy non-rendered (not everyone reads our documentation on GitHub). It also seems to go against the whole point of choosing and enforcing some sort of formatting.

Reply:
Agreed